src/main/java/org/example/bookstore/dto/userdto/UserRegistrationRequestDto.java

```diff
@@ -0,0 +1,29 @@
+package org.example.bookstore.dto.userdto;
+
+import jakarta.validation.constraints.AssertTrue;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+import org.hibernate.validator.constraints.Length;
+
+@Data
+public class UserRegistrationRequestDto {
+
+    @NotBlank
+    @Email
+    private String email;
+    @NotBlank
+    @Length(min = 4,max = 32)
+    private String password;
+    @NotBlank
+    @Length(min = 4,max = 32)
+    private String repeatPassword;
+    private String firstName;
+    private String lastName;
+    private String shippingAddress;
+
+    @AssertTrue(message = "Password and repeat password must match")
+    public boolean isPasswordMatch() {
+        return password.equals(repeatPassword);
+    }
```

Comment:
This logic should be implemented using annotation, please read requirement refarully